docs/developer/organization.md

```diff
@@ -0,0 +1,420 @@
+  calls the `WriteCheckpoint` function at an interval specified in the YAML
+  input file. This function is called in `RDySetup`.
+* `ReadCheckpointFile` (also defined in `checkpoint.c`) reads the checkpoint
+  file (in a format specified within the YAML input. This function is called
+  in `RDySetup` if the input file specifieѕ that a simulation should be
+  restarted.
+
+## Running Ensembles
+
+RDycore can run _ensembles_, which are sets of concurrently-running simulations
+("ensemble members") with variations in selected input parameters. RDycore uses
+an extremely simple mechanism to run these simulations within a single MPI job:
+
+* on initialization, RDycore splits its "world" communicator into a number of
+  smaller, communicators of equal size
+* each of these smaller communicators is assigned to a single ensemble member
+* each ensemble member is configured according to the ensemble specification
+  in the [YAML input file](../common/input.md)
+* all ensemble members are run concurrently to completion
+
+Currently, ensemble calculations only run ensemble members, possibly generating
+member-specific output. No post-processing or analysis has been implemented,
+though it would likely be simple to do so.
+
+### Ensemble member configuration
+
+The data for each ensemble member is listed explicitly in the [ensemble section](../common/input.md#ensemble)
+of the YAML input specification. All parameters for ensemble members override
+the corresponding parameters in other sections of the file. This logic is
+implemented in the `ConfigEnsembleMember` function within [ensemble.c](https://github.com/RDycore/RDycore/blob/main/src/ensemble.c).
+
+## Support for Fortran
+
+A Fortran version of the public C interface is available in an [`rdycore` Fortran module](https://github.com/RDycore/RDycore/blob/main/src/f90-mod/rdycore.F90)
+in the `src/f90-mod` directory. This module is hand-written and uses the
+[Fortran 2003 `iso_c_binding`](https://fortranwiki.org/fortran/show/iso_c_binding)
+standard intrinsic module to map C functions to equivalent Fortran subroutines
+and functions, and defines appropriate data types.
+
+The mapping from a C function to a Fortran subroutine (or function) is
+accomplished in two parts:
+
+1. A C function is made available to Fortran by defining a Fortran function
+   (or function) using a `bind(C)` annotation specifying the case-sensitive name
+   of the C function. All data types in this Fortran function must correspond to
+   supported C data types via the `iso_c_binding` module. The function returns
+   an integer corresponding to the `PetscErrorCode` return type of the C
+   function. We refer to such a function as a "C-bound Fortran function".
+
+2. A Fortran "wrapper" subroutine is defined that calls the C-bound Fortran
+   function defined in item 1. This subroutine follows the PETSc convention in
+   which the last argument (`ierr`) gets the return value of the C-bound Fortran
+   function.
```

Comment:
It might be worthwhile to give an example of a C function and a Fortran subroutine say
- `integer(c_int) function rdycreate_(...)`, and
- `subroutine RDyCreate(...)`